CMakeLists.txt

```diff
@@ -29,7 +29,7 @@ project(daphne-prototype LANGUAGES CXX C)
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 set(CMAKE_BUILD_WITH_INSTALL_NAME_DIR ON)
-set(CMAKE_CXX_STANDARD 17 CACHE STRING "C++ standard to conform to")
+set(CMAKE_CXX_STANDARD 20 CACHE STRING "C++ standard to conform to")
```

Comment:
Do we understand the consequences of changing the C++ standard for the DAPHNE use-cases?

Reply:
There was already some initial discussion with @pdamme in #694 regarding this. I think moving to a C++20-compliant compiler is fine as long as the documentation is clear about this. It might be a good idea to separate a potential switch to C++20 out of this PR and push it isolated so that we can (a) bring it in quicker and (b) revert if we realize that things break. Accordingly, the CI would need to be updated as well.

Re: Use Cases: Do we know anything about the environments where Daphne is run? Looking at our current dependency management I doubt that we do.